File: backend/src/Summary.kt
package com.mehtasankets.mmflow

data class Summary (
    val monthToDateSummary: SummaryData,
    val yearToDateSummary: SummaryData
)

data class SummaryData (
    val total: Double,
    val previousTotal: Double,
    val totalByCategory: Map<String, Double>,
    val totalByUser: Map<String, Double>
)

File: frontend/store/Summary.js
import { observable } from "mobx"

export default class Summary {
    @observable expenseSheetName
    @observable monthToDateSummary
    @observable yearToDateSummary

    constructor(expenseSheetName, monthToDateSummary, yearToDateSummary) {
        this.expenseSheetName = expenseSheetName
        this.monthToDateSummary = monthToDateSummary
        this.yearToDateSummary = yearToDateSummary
    }
}